src/flb_output.c

```diff
@@ -173,6 +173,98 @@ void flb_output_coro_add(struct flb_output_instance *ins, struct flb_coro *coro)
     mk_list_add(&out_flush->_head, &ins->flush_list);
 }
 
+/*
+ * Queue a task to be flushed at a later time
+ */
+static int flb_output_task_queue_enqueue(struct flb_task_queue *queue,
+                                         struct flb_task *task,
+                                         struct flb_output_instance *out_ins,
+                                         struct flb_config *config)
+{
+    struct flb_task_enqueued *queued_task;
+
+    queued_task = flb_malloc(sizeof(struct flb_task_enqueued));
+    if (!queued_task) {
+        flb_errno();
+        return -1;
+    }
+    queued_task->out_instance = out_ins;
+    queued_task->task = task;
+    queued_task->config = config;
+
+    mk_list_add(&queued_task->_head, &queue->pending);
+    return 0;
+}
+
+/*
+ * Pop task from pending queue and flush it
+ */
+static int flb_output_task_queue_flush_one(struct flb_task_queue *queue)
+{
+    struct flb_task_enqueued *queued_task;
+    int ret;
+
+    ret = mk_list_is_empty(&queue->pending);
+    if (ret == 0) {
+        flb_error("Attempting to pop task queue with no tasks enqueued");
+        return -1;
```

Comment:
Is this a good user experience for the log message- or will this just confuse users? 

Also, is the -1 return planned? If there is no pending task, should we take one from the queue instead of returning error?

Reply:
Sorry I realized pending means not started and the in_progess queue is for tasks that have started

Reply:
This log should never print. It's just for the functions completeness to do some input checking up front

Reply:
pending